### DISCUSSION TITLE
Question about VSG parameters change during the TDS

### DISCUSSION THREAD

**Discussion by xiaoheimai95** [CURENT/andes#452] [Q&A] [answered]
Dear Dr. Cui and Jinning,
We want to modify the value of `M` and `D`  of VSG unit during the TDS by the following code:
```
ss.TDS.config.tf = 2.0
ss.TDS.run()
ss.REGCV1.alter('M', REGCV1_10, 2)
ss.REGCV1.alter('D', REGCV1_10, 10)
ss.TDS.config.tf = 10
ss.TDS.run()
```
Can this method successfully modify the parameters of the VSG during the TDS process?

**Comment by jinningwang** [comment]
An easy way to test is that, after altering the parameter, using Model.get method to check if the value is changed.

**Comment by cuihantao** [comment, marked as answer]
Altering `D` should work as expected.

Altering `M` will not take effect in the current version. It is due to how time constants associated with differential equations are processed. You can tell from https://docs.andes.app/en/latest/groupdoc/RenGen.html#id16 that `M` is the time constant on the left side of differential equations `T x' = f(x, y)`.

At the very beginning of simulations, left-hand-side time constants of differential equations are arranged into `dae.Tf`. To modify `M`, for now, you can do it in two steps:

First, do

```
ss.REGCV1.alter('M', REGCV1_10, 2)
```
to alter the value and store its per-unit value in `ss.REGCV1.M.v`.

Next, modify `dae.Tf`:

```
ss.dae.Tf[ss.REGCV1.dw.a] = ss.REGCV1.M.v
```

Edit: index `dae.Tf` by the address of the state, not the parameter.

**Reply by cuihantao** [reply]
The issue has been fixed on my latest `develop` branch. `alter` should work as expected without having to manually update `dae.Tf`.

**Reply by xiaoheimai95** [reply]
So if I install the former version of `andes`, I should use the above method to set up `M`.  And if I update into the latest version of `andes-develop,` I just need to use the basic code `ss.REGCV1.alter('M', REGCV1_10, 2)` to set it up, right?

**Reply by cuihantao** [reply]
That is correct!

**Reply by xiaoheimai95** [reply]
Thanks a lot!
